openedx_lti_tool_plugin/views.py

```diff
@@ -429,11 +430,27 @@ def handle_ags(
         if AGS_SCORE_SCOPE not in ags_endpoint.get('scope', []):
             raise LtiToolLaunchException(_(f'Missing required AGS scope: {AGS_SCORE_SCOPE}'))
 
-        return LtiGradedResource.objects.get_or_create(
+        graded_resource = LtiGradedResource.objects.filter(
             lti_profile=lti_profile,
             context_key=context_key,
             lineitem=lineitem,
-        )
+        ).first()
+
+        if graded_resource:
+            return None
+
+        try:
+            graded_resource = LtiGradedResource(
+                lti_profile=lti_profile,
+                context_key=context_key,
+                lineitem=lineitem,
+            )
+            graded_resource.full_clean()
+            graded_resource.save()
+        except ValidationError as ex:
+            raise LtiToolLaunchException(_(ex.messages[0])) from ex
+
+        return None
```

Comment:
function is not returning nothing now, right?. @alexjmpb 
```suggestion
            return
   
        try:
            graded_resource = LtiGradedResource(
                lti_profile=lti_profile,
                context_key=context_key,
                lineitem=lineitem,
            )
            graded_resource.full_clean()
            graded_resource.save()
        except ValidationError as ex:
            raise LtiToolLaunchException(_(ex.messages[0])) from ex
```

Reply:
The linter complained when I tried to do this. do you think we should ignore that?

Reply:
@alexjmpb What do you think about doing this instead?

```
graded_resource_values = {
    'lti_profile': lti_profile,
    'context_key': context_key,
    'lineitem': lineitem,
}

try:
    LtiGradedResource.objects.get(**graded_resource_values)
except LtiGradedResource.DoesNotExist:
    try:
        graded_resource = LtiGradedResource(**graded_resource_values) 
        graded_resource.full_clean()
        graded_resource.save()
    except ValidationError as exc:
        raise LtiToolLaunchException(_(exc.messages[0])) from exc
```